crates/weaver_semconv/src/group.rs

```diff
@@ -220,6 +256,62 @@ impl GroupSpec {
     }
 }
 
+fn validate_any_value_examples(
+    errors: &mut Vec<Error>,
+    any_value: Option<&AnyValueSpec>,
+    group_id: &str,
+    path_or_url: &str,
+) -> WResult<(), Error> {
+    if let Some(value) = any_value {
+        let common_examples = &value.common().examples;
+        if let Some(examples) = common_examples {
+            match examples.validate_any_value(value, group_id, path_or_url) {
```

Comment:
@lquerel don't we have a helper to simplify this code?

Reply:
I'm not aware of any, but this function could be simplified as follows.

```rust
fn validate_any_value_examples(
    errors: &mut Vec<Error>,
    any_value: Option<&AnyValueSpec>,
    group_id: &str,
    path_or_url: &str,
) -> WResult<(), Error> {
    if let Some(value) = any_value {
        if let Some(examples) = &value.common().examples {
            match examples.validate_any_value(value, group_id, path_or_url) {
                WResult::Ok(_) => {}
                WResult::OkWithNFEs(_, errs) => errors.extend(errs),
                WResult::FatalErr(err) => return WResult::FatalErr(err),
            }
        } else {
            match value {
                AnyValueSpec::String { .. } | AnyValueSpec::Strings { .. } => {
                    errors.push(Error::InvalidAnyValueExampleError {
                        path_or_url: path_or_url.to_owned(),
                        group_id: group_id.to_owned(),
                        value_id: value.id(),
                        error: format!("This value is a {} but it does not contain any examples.",
                            if let AnyValueSpec::String { .. } = value { "string" } else { "string array" }),
                    });
                }
                _ => {}
            }
        }

        if let AnyValueSpec::Map { fields, .. } = value {
            for field in fields {
                if let WResult::FatalErr(err) =
                    validate_any_value_examples(errors, Some(field), group_id, path_or_url)
                {
                    return WResult::FatalErr(err);
                }
            }
        }
    }

    WResult::Ok(())
}
```